packages/lesswrong/lib/collections/comments/views.ts

```diff
@@ -188,6 +188,14 @@ Comments.addView("postLWComments", (terms: CommentsViewTerms) => {
   };
 })
 
+Comments.addView("profileRecentComments", (terms: CommentsViewTerms) => {
+  return {
+    selector: {deletedPublic: false},
+    options: {sort: {isPinnedOnProfile: -1, postedAt: -1}, limit: terms.limit || 5},
+  };
+})
+ensureIndex(Comments, augmentForDefaultView({ isPinnedOnProfile: -1, postedAt: -1 }))
```

Comment:
This will first show pinned comments, then comments that used to be pinned but have been unpinned, and then all comments that have never been pinned, because `isPinnedOnProfile === false` will be come before `isPinnedOnProfile === undefined`.

This is probably fixable with something like:
```
return {
  selector: {deletedPublic: false},
  syntheticFields: {isPinned: {$ifNull: ["$isPinnedOnProfile", false]}},
  options: {sort: {isPinned: -1, postedAt: -1}, limit: terms.limit || 5},
};
```
but we'll probably need to check the performance implications because that's now an `aggregate` instead of a simple `find`. I also have no idea whether that index will still be be useable or not (probably not).

Reply:
Thanks Ollie - forgot to add a migration to fill in the default values, now that's in so this should be good

Reply:
Ok, great!